tests/default/flow_framework/workflow/provision.yaml

```diff
@@ -32,4 +32,4 @@ chapters:
       payload:
         openai_key: '1234556'
     response:
-      status: 200
\ No newline at end of file
+      status: 202
```

Comment:
Line 3 speceifies version >=2.12.  This is a change for 3.0+.

I'm not sure how we are supposed to specify a version change.  Do we make line 3 `>=2.12 && <3` and have a new file for `>=3.0.0`?  

@Xtansia advice here? Developer guide doesn't cover major version API changes.

Reply:
@nhtruong don't we need to have `3.0.0-alpha-1` rather `3.0` here? That's why it's failing for 3.0

Reply:
It's getting much more complicated to host 2 different major versions of the spec (2.x and 3.x) and test them now. Having a branch for each major version (with `main` hosting the latest unreleased major version) will solve this issue.  What's your thought on this?
cc: @Xtansia @dblock

Reply:
I would avoid duplicating/branching the spec because it means backporting hell like we have in so many repos. You should be able to use `x-version-added` and friends to specify that in 2.x the API returns 200 and in 3.x it returns 202. You could also say that the API returns either of 200/202 with a comment, which I think is ok for now.

Reply:
Thanks @dblock! Pushed the change

Reply:
If we change the type to array instead of integer for expected response [here](https://github.com/opensearch-project/opensearch-api-specification/blob/main/json_schemas/test_story.schema.yaml#L193), that would mean changing all the responses in the test files to return an array rather an int?

Reply:
You can have `anyOf` in the schema between an integer and an array of integers.